### ISSUE TITLE
Sync (Super Slow)

### ISSUE THREAD

**Issue by WinMerge2021** [WinMerge/winmerge#771]
I have been a long time user of WinMerge and I thought version 2.14.0 was very good.  Never had an issue with it.  Ever since upgrading to the latest, I noticed this one issue and it could be my removable USB drive.  But when syncing files (even small size), it takes forever for it to complete the sync, like 3 minutes on a file that is only 5 kb.  **If I manually sync those files through Windows Explorer, than the process is instantaneous.**  This never has happened in version 2.14.0.

Windows 7 was bad but now I have upgraded to Windows 10, it is worst.  I have thought it was the virus protection and many other things but again, if I manually sync through Windows Explorer, then everything is instantaneous.

Any thoughts on this?

**Comment by sdottaka** [comment]
I tried copying from USB-DRIVE to C-Drive or copying in the opposite direction in the folder compare window, but I couldn't reproduce the very slow behavior.

Please let me know if you know the reproducible conditions.

**Comment by WinMerge2021** [comment]
Maybe the USB drives need to be optimized. Can you right click on your drive and check under properties if it is optimized? The external usb drive is optimized and has no issues while the usb thumb drive is not optimized and thus have the sync issue.

Thank you

**Comment by WinMerge2021** [comment]
So I found a solution.

I have an exact backup thumb drive with the same specs.

If the usb thumb drive is formatted NTFS, then the delay issue exist. If is formatted as exFat, then the is no issue when syncing in winmerge.  Again, Verizon 2.14 has no issue with this drive being NTFS. Has the copy/sync process changed? Thanks

**Comment by sdottaka** [comment]
Thank you for the info.

I tested it on a USB drive formatted with NTFS, but I couldn't reproduce the problem that the processing was very slow.

Certainly, the method of copying and deleting has changed from 2.14.

To address the issue at https://github.com/WinMerge/winmerge/issues/339, first, manipulate the file in the same way as 2.14 (SHFileOperation ()), and if that fails, use IFileOperation.

Do the files or folders you try to sync contain very long names?

**Comment by WinMerge2021** [comment]
Yes, they have long pathes but I think that is not the issue because when I was testing the zero byte file, it was in the root.

Furthermore, SHFileOperation api works because I used that in one of my other program, and no issue.

**Comment by WinMerge2021** [comment]
OK, a little bit more about this issue.  The two PNY Flash Drives that I have seems to NOT be compatible with Windows 10, especially when formatted as NTFS.  I bought two new flash drives today from another brand and they worked fine.

I don't know if it is a Windows 10 thing or a WinMerge thing but in case if you are curious, this is the drive that is giving me trouble. 

https://www.amazon.com/gp/product/B00JN1TOHM

These drives worked fine with the other version of WinMerge and WIndows 7.

Also, I will start a new thread but with the two new drives working, there is still a lag when syncing a big 10 gb file in WinMerge.  After it is done copying, the pointer is still in the circle thinking icon and will not release my USB Flash Drive.  If I copy it directly in Windows Explorer, then after it is done, I can eject my USB Flash Drive right away.

**Comment by sdottaka** [comment]
I got a USB drive (P-FD32GTBOP-GE) with a different capacity but a similar model number, so I tried it, but I couldn't reproduce it.

It seemed to be the same speed as copying in Explorer

**Comment by WinMerge2021** [comment]
I just want to confirm as of version 2.16.16, the lag after copying a large file issue has been resolved.  Thank you for fixing this.

How did you finally fix this issue because it was happening for the longest of time for 2.16..., but never in 2.14.0.

**Comment by sdottaka** [comment]
After copying the files, the various properties of the files (Modified Date, etc.) were being re-fetched, and this was the cause of the slowdown. Since the files were copied, the properties of the files were assumed to be the same as the original ones, so I stopped retrieving the properties.
When I first tried this, I didn't realize this because there were only a few files to copy.
